mllib/src/main/scala/org/apache/spark/ml/tree/Split.scala

```diff
@@ -189,3 +189,76 @@ class ContinuousSplit private[ml] (override val featureIndex: Int, val threshold
     OldSplit(featureIndex, threshold, OldFeatureType.Continuous, List.empty[Double])
   }
 }
+
+/**
+ * Split with pointers to the left and right child nodes in decision tree.
+ * Used to update the node index cache.
+ * @param split Input split
+ */
+class SplitWithChildNodeInfo private[ml] (split: Split) extends Split {
+  private var _leftChildIndex : Option[Int] = None
+  private var _rightChildIndex : Option[Int] = None
+
+  /**
+   * Create split with pointers to left and right child nodes
+   * @param split Input split
+   * @param leftChildIndex Index of left child node
+   * @param rightChildIndex Index of right child node
+   */
+  def this(split: Split, leftChildIndex: Int, rightChildIndex: Int) = {
```

Comment:
may be overloading the idea of a "split" by including the children.  I would suspect the node are responsible for children, the splits are responsible to for the decision to go left or right? 

Is this maneuver for performance?